generators/connection/index.js

```diff
@@ -116,7 +117,7 @@ module.exports = class ConnectionGenerator extends Generator {
   prompting () {
     this.checkPackage();
 
-    const databaseName = snakeCase(this.pkg.name);
+    const databaseName = validate(this.pkg.name).validForNewPackages ? this.pkg.name : snakeCase(this.pkg.name);
```

Comment:
This probably also should stay with `snakeCase` - many databases have very limiting restrictions on what is allowed for database names.